tabgpt/tabular_dataset.py

```diff
@@ -1,34 +1,69 @@
+from tabgpt.col_embed import add_positional_info, remove_index
 import torch
 import numpy as np
 from torch.utils.data import Dataset, ConcatDataset
 import os
 import logging
+from IPython import embed
+import json
+
 
 
 class TabularDataset(Dataset):
-    def __init__(self, folder_path):
+    def __init__(self, folder_path, task_description_path,n_targets):
         self.folder_path = folder_path
         self.file_names = os.listdir(folder_path)
         # Sort the file names by the extracted index
         self.sorted_files = sorted(self.file_names, key=self.extract_index)
 
+        self.n_targets = n_targets
+
+        with open(os.path.join(task_description_path,'task_description.json'), 'r') as f:
+            self.task_dict = json.load(f)
+
+        self.cache_files = {}
 
     def __len__(self):
-        return len(self.file_names)
+        return len(self.file_names) * self.n_targets
+
+    
+    def _load_file(self, file_path):
+        if self.cache_files:
+            # Load and cache the file if it's not already cached
+            if file_path not in self.file_cache:
+                data = torch.tensor(np.load(file_path))
+                self.file_cache[file_path] = data
+            return self.file_cache[file_path]
+        else:
+            # Load the image without caching
+            return torch.tensor(np.load(file_path))
 
     def __getitem__(self, idx):
-        file_path = os.path.join(self.folder_path, self.sorted_files[idx])
+        actual_idx = idx // self.n_targets
```

Comment:
we need to map back to original idx, so if there are k rows and 2 targets, idx goes to k*2